text/src/autogluon/text/automm/models/huggingface_text.py

```diff
@@ -53,12 +69,23 @@ def __init__(
                     - 'xlm-roberta-base'
         num_classes
             The number of classes. 1 for a regression task.
+        adaptation
+            Whether to adapt the pre-trained model. Currently supports: LoRA
         """
         super().__init__()
         logger.debug(f"initializing {checkpoint_name}")
         self.checkpoint_name = checkpoint_name
         self.num_classes = num_classes
         self.model = AutoModel.from_pretrained(checkpoint_name)
+        self.adaptation = adaptation
+        self.lora_r = lora_r
+        self.lora_alpha = lora_alpha
+
+        if self.adaptation == 'lora':
+            print('Applying LoRA adaptation to attention layers...')
+            replace_layers_with_lora(self.model, "query", self.lora_r, self.lora_alpha)
+            replace_layers_with_lora(self.model, "value", self.lora_r, self.lora_alpha)
+
```

Comment:
This layer replacement operations can be moved outside of specific model classes. We can move them to the end of `create_model`, i.e., after the model is created. In this way, we can apply lora to any models, not limited to text models.

Reply:
Good idea!